enhancements/sig-architecture/58-addon-configuration/README.md

```diff
@@ -0,0 +1,439 @@
+	// If it is not set, open-cluster-management-agent-addon namespace is used to install the add-on agent.
+	// +optional
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	InstallNamespace string `json:"installNamespace,omitempty"`
+
+	// configuration represents the configuration namespace and name for the current add-on.
+	// In scenario where the current add-on has its own configuration.
+	// +optional
+	Configuration ConfigurationReferent `json:"configuration,omitempty"`
+}
+
+// ConfigurationReferent represents the namespace and name for the current add-on.
+type ConfigurationReferent struct {
+	// namespace of the current add-on configuration.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// name of the current add-on configuration.
+	Name string `json:"name"`
+}
+
+// ManagedClusterAddOnStatus provides information about the status of the operator.
+// +k8s:deepcopy-gen=true
+type ManagedClusterAddOnStatus struct {
+	// Deprecated: Use configReference instead
+	// addOnConfiguration is a reference to configuration information for the add-on.
+	// This resource is use to locate the configuration resource for the add-on.
+	// +optional
+	AddOnConfiguration ConfigCoordinates `json:"addOnConfiguration",omitempty`
+
+	// configReference is a reference to the add-on configuration.
+	// +optional
+	ConfigReference ConfigReference `json:"configReference",omitempty`
+}
+
+// ConfigReference is a reference to the add-on configuration.
+// This resource is used to locate the configuration resource for the current add-on.
+type ConfigReference struct {
+	// group of the add-on configuration.
+	// This field is synced from ConfigCoordinates group.
+	Group string `json:"group"`
+
+	// version of the add-on configuration.
+	// This field is synced from ConfigCoordinates version.
+	Version string `json:"version"`
+
+	// resource of the add-on configuration.
+	// This field is synced from ConfigCoordinates resource.
+	Resource string `json:"resource"`
+
+	// namespace of the add-on configuration.
+	// This field is synced from ConfigCoordinates default namespace and ConfigurationReferent namespace.
+	// If both namespaces are defined, the ConfigurationReferent namespace will overwrite the ConfigCoordinates
+	// default namespace.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+
+	// name of the add-on configuration.
+	// This field is synced from ConfigCoordinates default name and ConfigurationReferent name.
+	// If both names are defined, the ConfigurationReferent name will overwrite the ConfigCoordinates
+	// default name.
+	Name string `json:"name"`
+
+	// lastObservedGeneration is the observed generation of the add-on configuration.
+	LastObservedGeneration int64 `json:"lastObservedGeneration"`
+}
+```
+
+As a specific case, we want to introduce a `AddOnDeployementConfig` API in cluster scope to configure the add-on
+deployment.
```

Comment:
please clarify that user can also use their own crd. I think an example of customized CRD will be helpful also.

Reply:
I will think the example

Reply:
done